kong/llm/drivers/shared.lua

```diff
@@ -168,6 +180,57 @@ local function handle_stream_event(event_table, model_info, route_type)
   end
 end
 
+local function complex_split(str, delimiter)
```

Comment:
I don't think it needs a unit test on a helper function. We know it's working because we test splitting up complex event strings.

I have annotated them though.